NumLib/Fem/Deformation/BMatrixPolicy.h

```diff
@@ -0,0 +1,139 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_FEM_BMATRIXPOLICY_H_
+#define NUMLIB_FEM_BMATRIXPOLICY_H_
+
+#include "NumLib/Fem/ShapeMatrixPolicy.h"
+
+namespace NumLib
+{
+/// Kelvin vector dimensions for given displacement dimension.
+template <int DisplacementDim>
+struct KelvinVectorDimensions;
+
+template <>
+struct KelvinVectorDimensions<1>
+{
+    static int const value = 1;
+};
+
+template <>
+struct KelvinVectorDimensions<2>
+{
+    static int const value = 4;
+};
+
+template <>
+struct KelvinVectorDimensions<3>
+{
+    static int const value = 6;
+};
+
+//
+// Kelvin vector and matrix templates for given displacement dimension.
+//
+
+template <int DisplacementDim>
+using KelvinVectorType =
+    Eigen::Matrix<double, KelvinVectorDimensions<DisplacementDim>::value, 1>;
+
+template <int DisplacementDim>
+struct KelvinMatrixTypeImpl
+{
+    using type =
+        Eigen::Matrix<double, KelvinVectorDimensions<DisplacementDim>::value,
+                      KelvinVectorDimensions<DisplacementDim>::value,
+                      Eigen::RowMajor>;
+};
+
+/// Special case for dimension one because of column major storage requirement
+/// from Eigen3.
+template <>
+struct KelvinMatrixTypeImpl<1>
+{
+    using type = Eigen::Matrix<double, 1, 1, Eigen::ColMajor>;
+};
+
+template <int DisplacementDim>
+using KelvinMatrixType = typename KelvinMatrixTypeImpl<DisplacementDim>::type;
+
+/// An implementation of ShapeMatrixPolicy using fixed size (compile-time) eigen
+/// matrices and vectors.
+template <typename ShapeFunction, unsigned DisplacementDim>
+struct EigenFixedBMatrixPolicy
+{
+    template <int N>
+    using _vectorType = typename ::detail::EigenMatrixType<N, 1>::type;
```

Comment:
I think typedefs should always start uppercase. So please put them into the private section if you don't want them to be used outside.